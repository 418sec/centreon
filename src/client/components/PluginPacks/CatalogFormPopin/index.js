export { default as CatalogFormPopin } from './CatalogFormPopinContainer';