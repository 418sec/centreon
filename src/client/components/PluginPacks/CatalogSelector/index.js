export { default as CatalogSelector } from './CatalogSelectorContainer';