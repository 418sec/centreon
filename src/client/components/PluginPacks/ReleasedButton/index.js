export { default as ReleasedButton } from './ReleasedButtonContainer';