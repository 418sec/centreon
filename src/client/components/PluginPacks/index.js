export { default as PluginPacksList } from './PluginPacksListContainer';