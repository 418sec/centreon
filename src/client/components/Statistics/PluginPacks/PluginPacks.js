import React from 'react'
import { withStyles } from 'material-ui/styles'
import { Bar } from 'react-chartjs-2'
import Typography from 'material-ui/Typography'
import TextField from 'material-ui/TextField'
import { CircularProgress } from 'material-ui/Progress'

const styles = theme => ({
  statisticsContainer: {
    display: 'flex',
    flexDirection: 'column'
  },
  statisticsHeader: {
    display: 'flex',
    flex: '0',
    flexDirection: 'row'
  },
  chartContainer: {
    position: 'relative',
    flex: 1,
    height: '85vh',
    width: '95%'
  },
  title: {
    flex: 1,
  },
  searchField: {
    paddingRight: '100px'
  },
  progress: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    zIndex: 1,
  },

})

const PluginPacks = ({ classes, slug, loading, options, data, searchValue, handleClick, handleSearch }) => (
  <div className={classes.statisticsContainer}>
    <div className={classes.statisticsHeader}>
      <Typography type='display1' align='center' className={classes.title}>
        {data.title}
      </Typography>
      {loading &&
        <CircularProgress className={classes.progress} size={50}/>
      }
      {!slug &&
        <TextField
          id="searchPP"
          label="Search"
          className={classes.searchField}
          value={searchValue}
          onChange={handleSearch}
          margin="normal"
        />
      }
    </div>
    <div className={classes.chartContainer}>
      <Bar
        //height={600}
        data={data}
        getElementAtEvent={handleClick}
        options={options}
      />
    </div>
  </div>
)

export default withStyles(styles)(PluginPacks)