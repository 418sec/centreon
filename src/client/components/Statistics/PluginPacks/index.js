export { default as PluginPacks } from './PluginPacksContainer';