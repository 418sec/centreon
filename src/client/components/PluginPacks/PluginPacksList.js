import React, { Component } from 'react'
import {
  PagingState,
  FilteringState,
  CustomPaging
} from '@devexpress/dx-react-grid'
import {
  Grid,
  Table,
  TableHeaderRow,
  TableFilterRow,
  PagingPanel
} from '@devexpress/dx-react-grid-material-ui'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import { TableRow, TableCell } from 'material-ui'
import { HeaderBar } from 'Components/HeaderBar'
import { CatalogFormPopin } from 'Components/PluginPacks/CatalogFormPopin'
import { CatalogSelector } from 'Components/PluginPacks/CatalogSelector'
import { ReleasedButton } from 'Components/PluginPacks/ReleasedButton'
import { FilterCell } from 'Components/PluginPacks/FilterCell'

const styles = theme => ({
  root: {
    padding: '0 15px',
    overflow: 'auto'
  },
  TableCell: {
    wordWrap: 'break-word',
    padding: '4px 14px',
  },
  check: {
    fill: '#88B917'
  },
  checkInactive: {
    fill: '#d1d2d4'
  },
  textField: {
    '&:hover': {
      background:'transparent'
    },
  }
})

class PluginPacksList extends Component {

  tableRowTemplate = ({children, row}) => {
    return (
      <TableRow id={row.slug}>
        {children}
      </TableRow>
    )
  }

  render = () => {
    const { columns, rows, itemsPerPage, totalItems, currentPage, classes } = this.props

    return (
      <section>
        <div id="pluginPackList" className={classes.root}>
          <Grid
            columns={columns}
            rows={rows}
          >
            <FilteringState
              onFiltersChange={this.props.changeFilters}
            />
            <PagingState
              currentPage={currentPage}
              onCurrentPageChange={this.props.changeCurrentPage}
              pageSize={itemsPerPage}
            />
            <CustomPaging
              totalCount={totalItems}
            />
            <Table
              rowComponent={this.tableRowTemplate}
              cellComponent={({ column, row, value }) => {
                if (column.name === 'released') {
                  return (
                    <TableCell>
                      <ReleasedButton
                        pluginPackId={row.id}
                      />
                    </TableCell>
                  )
                } else if (column.name === 'catalog_level') {
                  return (
                    <TableCell>
                      <CatalogSelector
                        id={row.id}
                        level={value}
                      />
                    </TableCell>
                  )
                }

                return (
                  <TableCell className={classes.TableCell}>{value}</TableCell>
                )
              }}
            />
            <TableHeaderRow />
            <PagingPanel id={"pagingPanel"}/>
            <TableFilterRow
              messages={{ filterPlaceholder: 'Filter' }}
              cellComponent={FilterCell}
            />
          </Grid>
        </div>
        <CatalogFormPopin />
      </section>
    )
  }
}

/**
 * Instance list Prototypes
 */
PluginPacksList.propTypes = {
  /**
   * Instance list css classes
   * @param {Object} classes
   */
  classes: PropTypes.object.isRequired,
}

PluginPacksList.defaultProps = {
  currentPage: 1
}

export default withStyles(styles)(PluginPacksList)
