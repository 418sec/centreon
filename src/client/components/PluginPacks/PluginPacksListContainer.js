import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import PluginPacksList from './PluginPacksList'
import { getPluginPacks } from 'Services/hubApi'
import { cleanHeader, replaceTitle} from 'Redux/header/headerActions'

class PluginPacksListContainer extends Component {

  constructor (props) {
    super(props)

    this.state = {
      columns: [
        {name: 'slug', title: 'Slug'},
        {name: 'name', title: 'Name'},
        {name: 'version', title: 'Version'},
        {name: 'catalog_level', title: 'Catalog'},
        {name: 'released', title: 'Released'},
      ],
      currentPage: 0,
      itemsPerPage: 10,
      filters: []
    }
  }

  componentDidMount = () => {
    const { cleanHeader, replaceHeaderTitle } = this.props
    cleanHeader()
    replaceHeaderTitle('Plugin Packs')

    this.getData()
  }

  changeFilters = (filters) => {
    this.setState(
      {
        filters: filters,
        currentPage: 0
      },
      this.getData
    )
  }

  changeCurrentPage = (currentPage) => {
    this.setState(
      {currentPage: currentPage},
      this.getData
    )
  }

  /**
   * example : page[number]=1&page[size]=30
   * @returns {string|*}
   */
  getPagination = () => {
    const { currentPage, itemsPerPage } = this.state
    return `?page[number]=${currentPage + 1}&page[size]=${itemsPerPage}`
  }

  /**
   * example : filter[instanceName]=centreon&filter[fingerprint]=abc
   * @returns {string|*}
   */
  getFilters = () => {
    const { filters } = this.state;

    return filters.reduce((acc, { columnName, value }) => {
      if (columnName == 'catalog_level' && value == '0') {
        acc.push(`filter[with_catalog]=false`)
      } else {
        acc.push(`filter[${columnName}]=${value}`)
      }
      return acc
    }, []).join('&')
  }

  getData = () => {
    const { token, displayPluginPacks } = this.props
    displayPluginPacks(this.getPagination(), this.getFilters(), token)
  }

  render = () => {
    const { columns, currentPage } = this.state
    const { items, itemsPerPage, totalPages, totalItems, ...props } = this.props

    const rows = items ?
      Object.keys(items).reduce((acc, key) => {
        acc[items[key].order] = items[key]
        return acc
      }, [])
      : []

    return (
      <PluginPacksList
        columns={columns}
        rows={rows}
        itemsPerPage={itemsPerPage}
        totalPages={totalPages}
        totalItems={totalItems}
        currentPage={currentPage}
        changeFilters={this.changeFilters}
        changeCurrentPage={this.changeCurrentPage}
        {...props}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    token: store.authReducer.token,
    items: store.pluginPacksReducer.items,
    totalPages: store.pluginPacksReducer.totalPages,
    totalItems: store.pluginPacksReducer.totalItems,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    cleanHeader: () => {
      return dispatch(cleanHeader())
    },
    replaceHeaderTitle: (title) => {
      return dispatch(replaceTitle(title))
    },
    displayPluginPacks: (pagination, filters, token) => {
      return dispatch(getPluginPacks(pagination, filters, token))
    },
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PluginPacksListContainer))