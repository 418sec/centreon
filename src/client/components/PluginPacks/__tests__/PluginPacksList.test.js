/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { PluginPacksList } from 'Components/PluginPacks'
import configureStore from 'redux-mock-store'
import createHistory from 'history/createBrowserHistory'


Enzyme.configure({ adapter: new Adapter() })

describe('PluginPacksList', () => {
  const mockStore = configureStore()
  let component

  it('renders correctly', () => {
    const store = mockStore({
      pluginPacksReducer: {
        items: {},
        catalogFormPopin: {
          open: false,
          id: null,
          level: null
        }
      }
    })

    component = shallow(
      <PluginPacksList.WrappedComponent
        store={store}
        cleanHeader={jest.fn()}
        replaceHeaderTitle={jest.fn()}
        displayPluginPacks={jest.fn()}
      />
    ).dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })

  describe('events', () => {
    const storeWithItems = mockStore({
      authReducer: {
        username: '',
        token: '',
      },
      pluginPacksReducer: {
        items: {
          '1': {
            order: 0,
            catalog_level: 1,
            id: 1,
            name: "pp 1",
            released: true,
            slug: "pp-1",
            version: "1.0.0"
          },
          '2': {
            order: 1,
            catalog_level: 2,
            id: 2,
            name: "pp 2",
            released: false,
            slug: "pp-2",
            version: "1.0.1"
          }
        },
        catalogFormPopin: {
          open: false,
          id: null,
          level: null
        },
        totalPages: 2,
        totalItems: 4,
      },
      confirmDialogReducer: {
        id: null,
        open: false,
        content: {
          body: '',
          cancel: '',
          confirm: '',
        },
        items: {}
      }
    })

    let cleanHeaderMock, replaceHeaderTitleMock, displayPluginPacksMock

    beforeEach(() => {
      cleanHeaderMock = jest.fn()
      replaceHeaderTitleMock = jest.fn()
      displayPluginPacksMock = jest.fn()
      const props = {
        token: '',
        items: {
          '1': {
            order: 0,
            catalog_level: 1,
            id: 1,
            name: "pp 1",
            released: true,
            slug: "pp-1",
            version: "1.0.0"
          },
          '2': {
            order: 1,
            catalog_level: 2,
            id: 2,
            name: "pp 2",
            released: false,
            slug: "pp-2",
            version: "1.0.1"
          }
        },
        totalPages: 2,
        totalItems: 4,
        cleanHeader: cleanHeaderMock,
        replaceHeaderTitle: replaceHeaderTitleMock,
        displayPluginPacks: displayPluginPacksMock
      }
      component = mount(
        <Provider store={storeWithItems}>
          <PluginPacksList.WrappedComponent
            {...props}
          />
        </Provider>
      )
    })

    it('filters on slug', () => {
      component.find('th#filterSlug input').simulate('change', {target: { value: 'pp' }})
      expect(component.childAt(0).instance().state.filters).toEqual([ { columnName: 'slug', value: 'pp' } ])
      expect(displayPluginPacksMock.mock.calls.length).toBe(2)
      expect(displayPluginPacksMock.mock.calls[1][1]).toEqual('filter[slug]=pp')
    })

    it('filters on name', () => {
      component.find('th#filterName input').simulate('change', {target: { value: 'network cisco' }})
      expect(component.childAt(0).instance().state.filters).toEqual([ { columnName: 'name', value: 'network cisco' } ])
      expect(displayPluginPacksMock.mock.calls.length).toBe(2)
      expect(displayPluginPacksMock.mock.calls[1][1]).toEqual('filter[name]=network cisco')
    })

    it('filters on catalog', () => {
      component.find('th#filterCatalog div[role="button"]').simulate('click')
      component.find('li#filterCatalog2').simulate('click')
      expect(component.childAt(0).instance().state.filters).toEqual([ { columnName: 'catalog_level', value: '2' } ])
      expect(displayPluginPacksMock.mock.calls.length).toBe(2)
      expect(displayPluginPacksMock.mock.calls[1][1]).toEqual('filter[catalog_level]=2')
    })

    it('filters on released', () => {
      component.find('th#filterReleased div[role="button"]').simulate('click')
      component.find('li#filterReleasedReleased').simulate('click')
      expect(component.childAt(0).instance().state.filters).toEqual([ { columnName: 'released', value: 'true' } ])
      expect(displayPluginPacksMock.mock.calls.length).toBe(2)
      expect(displayPluginPacksMock.mock.calls[1][1]).toEqual('filter[released]=true')
    })

    it('changes page', () => {
      /* TODO */
    })
  })
})