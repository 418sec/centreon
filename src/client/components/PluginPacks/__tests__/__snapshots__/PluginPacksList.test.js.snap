// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginPacksList renders correctly 1`] = `
<section>
  <div
    className="PluginPacksList-root-1"
    id="pluginPackList"
  >
    <Grid$$1
      columns={
        Array [
          Object {
            "name": "slug",
            "title": "Slug",
          },
          Object {
            "name": "name",
            "title": "Name",
          },
          Object {
            "name": "version",
            "title": "Version",
          },
          Object {
            "name": "catalog_level",
            "title": "Catalog",
          },
          Object {
            "name": "released",
            "title": "Released",
          },
        ]
      }
      rows={Array []}
    >
      <FilteringState
        defaultFilters={Array []}
        onFiltersChange={[Function]}
      />
      <PagingState
        currentPage={0}
        defaultCurrentPage={0}
        defaultPageSize={10}
        onCurrentPageChange={[Function]}
      />
      <CustomPaging
        totalCount={0}
      />
      <Table$$1
        cellComponent={[Function]}
        messages={Object {}}
        rowComponent={[Function]}
      />
      <TableHeaderRow$$1
        messages={Object {}}
      />
      <PagingPanel$$1
        id="pagingPanel"
        messages={Object {}}
      />
      <TableFilterRow$$1
        cellComponent={[Function]}
        messages={
          Object {
            "filterPlaceholder": "Filter",
          }
        }
      />
    </Grid$$1>
  </div>
  <Connect(CatalogFormPopinContainer) />
</section>
`;
